src/Adapter/Presenter/AbstractLazyArray.php

```diff
@@ -234,15 +249,33 @@ public function __clone()
     public function offsetGet($index)
     {
         if (isset($this->arrayAccessList[$index])) {
-            // if the index is associated with a method, execute the method an cache the result
-            if ($this->arrayAccessList[$index]['type'] === 'method') {
-                if (!isset($this->methodCacheResults[$index])) {
-                    $methodName = $this->arrayAccessList[$index]['value'];
-                    $this->methodCacheResults[$index] = $this->{$methodName}();
-                }
-                $result = $this->methodCacheResults[$index];
-            } else { // if the index is associated with a value, just return the value
-                $result = $this->arrayAccessList[$index]['value'];
+            $type = $this->arrayAccessList[$index]['type'];
+            switch ($type) {
+                case 'method':
+                    $isResultAvailableInCache = (isset($this->methodCacheResults[$index]));
```

Comment:
```suggestion
                    $isResultAvailableInCache = isset($this->methodCacheResults[$index]);
```

Reply:
Is this variable really needed, can be use directly in the `if` like on line 251 :thinking:

Reply:
When I modified this code, I was unsure about what `methodCacheResults` was. Was it a caching array ? Was it some methods name related to caching ? So I checked what it was and added this variable in order to make crystal clear for the next developer who will read this file.

Reply:
Also I personally like adding `(` and `)` around booleans statements 😛 so I guess it's a personal preference, not related to whether or not this property is valid.

Reply:
No problem, do what you think is the best for you :heart: